Make a meta form for user edit and custom data

esoergel opened the pull request:
I still want to do a little more testing on this before merging, but I'd like to get it in for code review and the build now.
:blowfish: :heavy_multiplication_x: :blowfish: 
@nickpell

esoergel committed:
Drop clean_commcare_user signal

esoergel committed:
Set location directly CommCareUser.create

esoergel committed:
Allow each user view to instantiate it's own form

esoergel committed:
Make a meta form for user edit and custom data

This should as-yet have no functional changes, but is necessary for some
eNikshay customizations that are forthcoming

nickpell commented:
```
======================================================================

ERROR: corehq.apps.users.tests.test_views:TestMobileWorkerListView.test_create_anonymous_mobile_worker

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched

    return func(*args, **keywargs)

  File "/mnt/commcare-hq-ro/corehq/apps/users/tests/test_views.py", line 67, in test_create_anonymous_mobile_worker

    'is_anonymous': True,

  File "/mnt/commcare-hq-ro/corehq/apps/users/tests/test_views.py", line 37, in _remote_invoke

    HTTP_X_REQUESTED_WITH='XMLHttpRequest',

  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 541, in post

    secure=secure, **extra)

  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 343, in post

    secure=secure, **extra)

  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 409, in generic

    return self.request(**r)

  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 494, in request

    six.reraise(*exc_info)

  File "/vendor/lib/python2.7/site-packages/django/core/handlers/exception.py", line 42, in inner

    response = get_response(request)

  File "/vendor/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response

    response = self.process_exception_by_middleware(e, request)

  File "/vendor/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response

    response = wrapped_callback(request, *callback_args, **callback_kwargs)

  File "/vendor/lib/python2.7/site-packages/django/views/generic/base.py", line 68, in view

    return self.dispatch(request, *args, **kwargs)

  File "/mnt/commcare-hq-ro/corehq/apps/style/decorators.py", line 17, in _wrapped

    return view_func(class_based_view, request, *args, **kwargs)

  File "/mnt/commcare-hq-ro/corehq/apps/style/decorators.py", line 53, in _wrapped

    return view_func(class_based_view, request, *args, **kwargs)

  File "/vendor/lib/python2.7/site-packages/django/utils/decorators.py", line 67, in _wrapper

    return bound_func(*args, **kwargs)

  File "/mnt/commcare-hq-ro/corehq/apps/domain/decorators.py", line 103, in _inner

    return view_func(req, domain_name, *args, **kwargs)

  File "/mnt/commcare-hq-ro/corehq/apps/users/decorators.py", line 21, in _inner

    return view_func(request, domain, *args, **kwargs)

  File "/vendor/lib/python2.7/site-packages/django/utils/decorators.py", line 63, in bound_func

    return func.__get__(self, type(self))(*args2, **kwargs2)

  File "/mnt/commcare-hq-ro/corehq/apps/users/views/mobile/users.py", line 515, in dispatch

    return super(MobileWorkerListView, self).dispatch(*args, **kwargs)

  File "/vendor/lib/python2.7/site-packages/django/utils/decorators.py", line 67, in _wrapper

    return bound_func(*args, **kwargs)

  File "/mnt/commcare-hq-ro/corehq/apps/domain/decorators.py", line 103, in _inner

    return view_func(req, domain_name, *args, **kwargs)

  File "/vendor/lib/python2.7/site-packages/django/utils/decorators.py", line 63, in bound_func

    return func.__get__(self, type(self))(*args2, **kwargs2)

  File "/mnt/commcare-hq-ro/corehq/apps/domain/views.py", line 238, in dispatch

    return super(LoginAndDomainMixin, self).dispatch(*args, **kwargs)

  File "/vendor/lib/python2.7/site-packages/django/views/generic/base.py", line 88, in dispatch

    return handler(request, *args, **kwargs)

  File "/vendor/lib/python2.7/site-packages/djangular/views/mixins.py", line 102, in post

    response_data = handler(in_data)

  File "/mnt/commcare-hq-ro/corehq/apps/users/views/mobile/users.py", line 736, in create_mobile_worker

    couch_user = self._build_anonymous_commcare_user()

  File "/mnt/commcare-hq-ro/corehq/apps/users/views/mobile/users.py", line 767, in _build_anonymous_commcare_user

    location=SQLLocation.objects.get(location_id=location_id),

  File "/vendor/lib/python2.7/site-packages/django/db/models/manager.py", line 85, in manager_method

    return getattr(self.get_queryset(), name)(*args, **kwargs)

  File "/vendor/lib/python2.7/site-packages/django/db/models/query.py", line 385, in get

    self.model._meta.object_name

DoesNotExist: SQLLocation matching query does not exist.

-------------------- >> begin captured logging << --------------------

auditcare.middleware: DEBUG: Middleware is running in a test context, disabling monkeypatch

--------------------- >> end captured logging << ---------------------
```

esoergel committed:
These might not be specified

esoergel committed:
Stop pretending cc-user and web user forms are the same

Shoulda done this earlier.  Honestly I wish these were just two different views
that maybe called the same functions, but this whole inheritance thing just
makes it needlessly complicated.

esoergel committed:
Disable tests of disabled functionality

I got rid of the signal, so of course these fail.  Leaving them in as comments
though, to later be reproduced in inheritance form

esoergel commented:
@nickpell I'm happy with where this is at now, should be g2g!
